lib/src/utils/window_main_state_listener.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:macos_ui/macos_ui.dart';
+
+/// A class that listens for changes to the application’s main window.
+///
+/// A common use-case for responding to such changes would be to mute the colors
+/// of certain primary UI elements when the window is no longer in focus, which
+/// is something native macOS applications do out of the box.
+///
+/// Example using [StreamBuilder]:
+///
+/// ```dart
+/// StreamBuilder(
+///   stream: WindowMainStateListener.instance.onChangedStream,
+///     builder: (context, _) {
+///       final bool isMainWindow
+///           = WindowMainStateListener.instance.isMainWindow;
+///
+///       return SomeWidget(
+///         isMainWindow: isMainWindow,
+///         child: …
+///       );
+///     },
+///  );
+/// ```
+class WindowMainStateListener {
```

Comment:
It occurs to me that with the eventuality of official multi-window support that calling this a window "main" state listener probably won't be correct; `WindowStateListener` will probably be more correct. Not an immediate issue, but if we correct for it now we'll have less work to do in the future.

Reply:
I mean, it will still listen to the windows’ main state, it will just need to be updated such that it can handle multiple windows, so I think the name is fine.

Reply:
When sub windows are in focus, are the considered the main window? I guess that would be the determining factor.

Reply:
I’m fairly certain that the focused window is the main one.

Reply:
Then I suppose we need to determine what a window's "main" state is